File: works/2436.cpp
#include <iostream>
#include <cmath>
#include <algorithm>
using namespace std;

long long GCD(long long a, long long b)
{
    if(b == 0) return a;
    return GCD(b, a%b);
}

int main()
{
    // 최대공약수, 최소공배수
    long long gcd, lcm;
    cin >> gcd >> lcm;
    long long a, b;

    lcm /= gcd;
    for(int i = 1; i * i <= lcm; i++)
    {
        if(lcm % i != 0) continue;
        long long a_tmp = i;
        long long b_tmp = lcm / i;
//        cout << a_tmp << ' ' << b_tmp << endl;
        // 최대공약수가 1
        if(GCD(a_tmp, b_tmp) == 1)
        {
            a = a_tmp;
            b = b_tmp;
        }
    }

    cout << a*gcd << ' ' << b*gcd;
}

File: works/2436.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun GCD(a:Long, b:Long): Long
{
    if(b == 0L) return a
    return GCD(b, a%b)
}

fun main()
{
    val br = BufferedReader(InputStreamReader(System.`in`))
    val _s = br.readLine().split(' ')
    var gcd = _s[0].toLong()
    var lcm = _s[1].toLong()
    var a = 0L
    var b = 0L

    lcm /= gcd
    var i = 0L
    while(true)
    {
        i++
        if(i*i > lcm) break
        if(lcm % i != 0L) continue
        val a_tmp = i
        val b_tmp = lcm / i

        if(GCD(a_tmp, b_tmp) == 1L)
        {
            a = a_tmp
            b = b_tmp
        }
    }
    val _a = a * gcd
    val _b = b * gcd
    print("$_a $_b")
}